### PR TITLE
Remove HybridBitSet::dummy

### PR DESCRIPTION
**Pull request by ljedrz**
This simplifies some of the `HybridBitSet` code.

cc @nnethercote

### CODE REVIEW

**Review by matthewjasper — COMMENTED** [review]
Nice!

**Comment by ljedrz** [comment]
@matthewjasper comment addressed.

File: src/librustc_data_structures/bit_set.rs [hunk 1]
```diff
@@ -525,33 +511,26 @@ impl<T: Idx> HybridBitSet<T> {
...
-                                let mut new_dense = self_sparse.to_dense();
-                                let changed = new_dense.union(other_dense);
-                                *self = HybridBitSet::Dense(new_dense);
-                                changed
-                            }
-                            _ => unreachable!()
-                        }
+                        let mut new_dense = self_sparse.to_dense();
+                        let changed = new_dense.union(other_dense);
+                       *self = HybridBitSet::Dense(new_dense);
```
Comment (nnethercote) [inline src/librustc_data_structures/bit_set.rs]: The indentation on this line is wrong.

File: src/librustc_data_structures/bit_set.rs [hunk 2]
```diff
@@ -487,18 +478,13 @@ impl<T: Idx> HybridBitSet<T> {
...
-                        changed
-                    }
-                    _ => unreachable!()
-                }
+                let mut dense = sparse.to_dense();
+                let changed = dense.insert(elem);
+                assert!(changed);
+                *self = HybridBitSet::Dense(dense);
+                changed
             }
```
Comment (nnethercote) [inline src/librustc_data_structures/bit_set.rs]: Might as well remove this blank line, now.

**Comment by nnethercote** [comment]
I tried this patch with an oldish (~3 weeks ago) repo and got compile errors, and then I updated and now it works. I figure NLL must have been enabled for rustc in the meantime?

**Comment by matthewjasper** [comment]
Yes, it got enabled with the beta upgrade

**Comment by nnethercote** [comment]
Cool, I didn't know it would improve this code. That's nice!

**Comment by ljedrz** [comment]
Whitespaces fixed.

**Pull request merged**